acrobat/blocks/app-banner/app-banner.js

```diff
@@ -0,0 +1,128 @@
+import { createTag } from "../../scripts/miloUtils.js";
+
+var link = '';
+var mobileOS = getMobileOperatingSystem();
+
+
+export default async function init(el) {
+  //Exit if not mobile/tablet 
+  if (mobileOS == 'unknown') {
+    el.innerHTML = '';
+    return;
+  }
+
+  createTag.then((createTag) => {
+    const bannerPlaceHolder = createTag('div', { class: 'placeholder-banner' }, '');
+    document.body.prepend(bannerPlaceHolder);
+    const children = el.querySelectorAll(':scope > div');
+    if (mobileOS === 'Android') {
+      link = children[4].textContent.trim();
+    } else if (mobileOS === 'iOS') {
+      link = children[7].textContent.trim();
+    }
+    const appBanner = createTag('div', { class: 'app-banner' }, '');
+    //Close banner button on the left, also the icon
+    const closeBtn = createTag('div', { class: 'app-banner-close', role: 'text', 'aria-label': 'Close banner' }, '×');
+    closeBtn.addEventListener('click', closeBanner);
+    const appBannerContent = createTag('div', { class: 'app-banner-content' }, '');
+    appBannerContent.append(closeBtn, getDecoratedBannerLeft(createTag, children),
+      getDecoratedAppDetails(createTag, children), getDecoratedBannerRight(createTag, children));
+    appBanner.append(appBannerContent);
+    el.innerHTML = '';
+    appBanner.style.display = 'block';
+    bannerPlaceHolder.remove();
+    document.body.classList.add('has-app-banner');
+    document.body.prepend(appBanner);
+  });
+}
+
+function openLink() {
+  window.open(link, "_blank");
+  closeBanner();
+}
+
+function closeBanner() {
+  const appBanner = document.querySelector('.app-banner');
+  document.body.classList.remove('has-app-banner');
+  appBanner.remove();
+}
+
+function getDecoratedBannerRight(createTag, children) {
+  const openText = children[3].textContent.trim();
+  const appBannerRight = createTag('div', { class: 'app-banner-right' }, '');
+  const openButton = createTag('div', { class: 'app-banner-button', role: 'text', 'aria-label': openText }, openText);
+  appBannerRight.addEventListener('click', openLink);
+  appBannerRight.append(openButton);
+  return appBannerRight;
+}
+
+function getDecoratedAppDetails(createTag, children) {
+  let rating = 0;
+  let reviews = 0;
+
+  if (mobileOS === 'Android') {
+    rating = children[5].textContent.trim();
+    reviews = children[6].textContent.trim();
+  } else if (mobileOS === 'iOS') {
+    rating = children[8].textContent.trim();
+    reviews = children[9].textContent.trim();
+  }
+
+  const appTitle = createTag('div', { class: 'app-banner-title' }, children[0].textContent.trim());
```

Comment:
For title we should use one of `h` tags, and for the text below we should use `p` tags

Reply:
The branch.io CSS style messes up if I add other tags instead of divs.

Reply:
Done